Description: adjust kpatch-build directories
 dkms puts modules into a different location than is expected, change
 paths to adjust for this. In addition the debian packaging uses /usr/lib
 instead of /usr/libexec for internal binaries.
Author: Chris J Arges <chris.j.arges@canonical.com>
Last-Update: 2022-05-24

--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -243,9 +243,9 @@
 		TOOLSDIR="$SCRIPTDIR"
 		DATADIR="$(readlink -f "$SCRIPTDIR/../kmod")"
 		PLUGINDIR="$(readlink -f "$SCRIPTDIR/gcc-plugins")"
-	elif [[ -e "$SCRIPTDIR/../libexec/kpatch/create-diff-object" ]]; then
+	elif [[ -e "$SCRIPTDIR/../lib/kpatch/create-diff-object" ]]; then
 		# installation path
-		TOOLSDIR="$(readlink -f "$SCRIPTDIR/../libexec/kpatch")"
+		TOOLSDIR="$(readlink -f "$SCRIPTDIR/../lib/kpatch")"
 		DATADIR="$(readlink -f "$SCRIPTDIR/../share/kpatch")"
 		PLUGINDIR="$TOOLSDIR"
 	else
@@ -258,12 +258,12 @@
 	if [[ -e "$SCRIPTDIR/create-diff-object" ]]; then
 		# git repo
 		SYMVERSFILE="$DATADIR/core/Module.symvers"
-	elif [[ -e "$SCRIPTDIR/../libexec/kpatch/create-diff-object" ]]; then
+	elif [[ -e "$SCRIPTDIR/../lib/kpatch/create-diff-object" ]]; then
 		# installation path
 		if [[ -e "$SCRIPTDIR/../lib/kpatch/$ARCHVERSION/Module.symvers" ]]; then
 			SYMVERSFILE="$(readlink -f "$SCRIPTDIR/../lib/kpatch/$ARCHVERSION/Module.symvers")"
-		elif [[ -e /lib/modules/$ARCHVERSION/extra/kpatch/Module.symvers ]]; then
-			SYMVERSFILE="$(readlink -f "/lib/modules/$ARCHVERSION/extra/kpatch/Module.symvers")"
+		elif [[ -e /lib/modules/$ARCHVERSION/updates/dkms/kpatch.Module.symvers ]]; then
+			SYMVERSFILE="$(readlink -f "/lib/modules/$ARCHVERSION/updates/dkms/kpatch.Module.symvers")"
 		fi
 	fi
 
